tests for diagnostics in callable `msg` in `torch.testing.assert_close`

pmeier opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #61031 remove UsageError in favor of ValueError
* #60639 change language from array to tensor
* #60638 relax type equality constraint
* #60637 Don't check stride by default
* #60636 Use maximum of tolerances in case of mismatching dtypes
* #60536 upcast to most precise dtype within their category before the comparison
* **#60254 tests for diagnostics in callable `msg` in `torch.testing.assert_close`**
* #60163 update docstring examples of `torch.testing.assert_close`
* #60091 Improve error messages of `torch.testing.assert_close` in case of mismatching values

Before we only tested that the correct error message is returned if `msg` is passed as callable. This adds tests that make sure that

- the inputs passed to the callable are the same inputs passed to `torch.assert_close` and
- the `diagnostics` namespace has the same attributes and types as documented.

Differential Revision: [D29556354](https://our.internmc.facebook.com/intern/diff/D29556354)

pmeier committed:
tests for diagnostics in callable `msg` in `torch.testing.assert_close`

Before we only tested that the correct error message is returned if `msg` is passed as callable. This adds tests that make sure that

- the inputs passed to the callable are the same inputs passed to `torch.assert_close` and
- the `diagnostics` namespace has the same attributes and types as documented.

[ghstack-poisoned]

pmeier committed:
Update on "tests for diagnostics in callable `msg` in `torch.testing.assert_close`"


Before we only tested that the correct error message is returned if `msg` is passed as callable. This adds tests that make sure that

- the inputs passed to the callable are the same inputs passed to `torch.assert_close` and
- the `diagnostics` namespace has the same attributes and types as documented.

[ghstack-poisoned]

pmeier committed:
Update on "tests for diagnostics in callable `msg` in `torch.testing.assert_close`"


Before we only tested that the correct error message is returned if `msg` is passed as callable. This adds tests that make sure that

- the inputs passed to the callable are the same inputs passed to `torch.assert_close` and
- the `diagnostics` namespace has the same attributes and types as documented.

[ghstack-poisoned]

pmeier committed:
Update on "tests for diagnostics in callable `msg` in `torch.testing.assert_close`"


Before we only tested that the correct error message is returned if `msg` is passed as callable. This adds tests that make sure that

- the inputs passed to the callable are the same inputs passed to `torch.assert_close` and
- the `diagnostics` namespace has the same attributes and types as documented.

[ghstack-poisoned]

pmeier committed:
Update on "tests for diagnostics in callable `msg` in `torch.testing.assert_close`"


Before we only tested that the correct error message is returned if `msg` is passed as callable. This adds tests that make sure that

- the inputs passed to the callable are the same inputs passed to `torch.assert_close` and
- the `diagnostics` namespace has the same attributes and types as documented.

[ghstack-poisoned]

pmeier committed:
Update on "tests for diagnostics in callable `msg` in `torch.testing.assert_close`"


Before we only tested that the correct error message is returned if `msg` is passed as callable. This adds tests that make sure that

- the inputs passed to the callable are the same inputs passed to `torch.assert_close` and
- the `diagnostics` namespace has the same attributes and types as documented.

[ghstack-poisoned]

mruberry reviewed (approved):
"diagnostics" is a better name

pmeier committed:
Update on "tests for diagnostics in callable `msg` in `torch.testing.assert_close`"


Before we only tested that the correct error message is returned if `msg` is passed as callable. This adds tests that make sure that

- the inputs passed to the callable are the same inputs passed to `torch.assert_close` and
- the `diagnostics` namespace has the same attributes and types as documented.

[ghstack-poisoned]

pmeier committed:
Update on "tests for diagnostics in callable `msg` in `torch.testing.assert_close`"


Before we only tested that the correct error message is returned if `msg` is passed as callable. This adds tests that make sure that

- the inputs passed to the callable are the same inputs passed to `torch.assert_close` and
- the `diagnostics` namespace has the same attributes and types as documented.

[ghstack-poisoned]

pmeier committed:
Update on "tests for diagnostics in callable `msg` in `torch.testing.assert_close`"


Before we only tested that the correct error message is returned if `msg` is passed as callable. This adds tests that make sure that

- the inputs passed to the callable are the same inputs passed to `torch.assert_close` and
- the `diagnostics` namespace has the same attributes and types as documented.

[ghstack-poisoned]

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29556354).
